Sources/TabBar/View/TabBar.swift

```diff
@@ -25,135 +25,130 @@ import SwiftUI
 
 /**
  `TabBar` – highly customizable tab bar for your SwiftUI application.
- 
+
  By using this component you will be able to add a view that
  switches between multiple child views using interactive user
  interface elements.
- 
+
  `TabBar` can be easily customized. You have to conform
  to `TabBarStyle` and `TabItemStyle` to customize bar
  and item respectively. To apply customization you have to inject
  them to tab bar using `tabBar(style:)` for bar
  and `tabItem(style:)` for item.
- 
+
  Usage:
- 
+
  ```
  TabBar(selection: $selection) { }
     .tabBar(style: CustomTabBarStyle())
     .tabItem(style: CustomTabItemStyle())
  ```
  */
 public struct TabBar<TabItem: Tabbable, Content: View>: View {
-    
-    private let selectedItem: TabBarSelection<TabItem>
-    private let content: Content
-    
-    private var tabItemStyle : AnyTabItemStyle
-    private var tabBarStyle  : AnyTabBarStyle
-    
-    @State private var items: [TabItem]
-    
+    @StateObject private var selectedItem: TabBarSelection<TabItem>
+    private let content: () -> Content
```

Comment:
In fact, delayed closures perform better in practice. SwiftUI needs to construct `View` instances to compare function signatures, resulting in a higher frequency of View initialization compared to the` View` body getter.

Reply:
Storing closures in a parameter makes the `View` (in this case `TabBar`) to re-evaluate on every `@State` change of the parent, even on unrelated one, because closures are not `Equatable` and `SwiftUI` cannot check if the return value of this closure changed or not.